src/core/Akka.Remote/Transport/DotNetty/TcpTransport.cs

```diff
@@ -193,7 +195,41 @@ public override void Disassociate()
             _channel.CloseAsync();
         }
     }
-    
+
+    internal sealed class NonBatchingTcpAssociationHandle : AssociationHandle
```

Comment:
Separated the association handles into batching / non-batching modes - so the offending code that caused #4434 never gets a chance to run when batching is disabled.